src/Services/InventoryUpdateService.php

```diff
@@ -37,319 +54,524 @@ class InventoryUpdateService
-    $itemMappingMethod = $configHelper->getItemMappingMethod();
-
-    $loggerContract->debug(
-      TranslationHelper::getLoggerKey(self::LOG_KEY_INVENTORY_UPDATE_START),
-      [
-        'additionalInfo' => ['fullInventory' => (string) $fullInventory],
-        'method' => __METHOD__
-      ]
-    );
+    try {
 
-    $page = 0;
-    $inventorySaveTotal = 0;
-    $inventorySaveSuccess = 0;
-    $inventorySaveFail = 0;
-    $saveInventoryDuration = 0;
-    $savedInventoryDuration = 0;
+      if ($this->syncIsBlocked($fullInventory)) {
+        throw new InventorySyncBlockedException("Another inventory sync is in progress, preventing this one from starting");
+      }
 
-    $errorMessage = null;
-    $amtOfDtosForPage = 0;
+      if (!$fullInventory) {
+        $lastGoodFullSyncStart = $this->statusService->getLastSuccessfulAttemptTime(true);
+        if (!isset($lastGoodFullSyncStart) || empty($lastGoodFullSyncStart)) {
+          $this->logger->info(
+            TranslationHelper::getLoggerKey(self::LOG_KEY_NO_SYNCS),
+            [
+              'additionalInfo' => [],
+              'method' => __METHOD__
+            ]
+          );
 
-    try {
-      $fields = $this->getResultFields();
-      /* Page size is tuned for a balance between memory usage (in plentymarkets) and number of transactions  */
-      $fields['itemsPerPage'] = AbstractConfigHelper::INVENTORY_ITEMS_PER_PAGE;
-      $variationSearchRepository->setFilters($this->getFilters($fullInventory));
+          // NOTICE: due to recursion, this log will show up AFTER the logs for the Full Sync!
+          $externalLogs->addInfoLog("Forced a full inventory sync as there are no successful full syncs on record");
+
+          return $this->sync(true, $manual);
+        }
+      }
+
+      // look up item mapping method at this level to ensure consistency and improve efficiency
+      $itemMappingMethod = $this->configHelper->getItemMappingMethod();
+
+      // TODO: remove dependency on VariationSearchRepositoryContract by replacing with a Wayfair wrapper
+      /** @var VariationSearchRepositoryContract */
+      $variationSearchRepository = pluginApp(VariationSearchRepositoryContract::class);
 
       // stock buffer should be the same across all pages of inventory
-      $stockBuffer = self::getNormalizedStockBuffer($configHelper, $loggerContract);
+      $stockBuffer = $this->getNormalizedStockBuffer();
+
+      $searchParameters = $this->getDefaultSearchParameters();
+
+      $pageNumber = 0;
+      $amtOfDtosForPage = 0;
+
+      $logKeyStart = self::LOG_KEY_START_PARTIAL;
+      if ($fullInventory) {
+        $logKeyStart = self::LOG_KEY_START_FULL;
+      }
+
+      $this->logger->info(TranslationHelper::getLoggerKey($logKeyStart), [
+        'additionalInfo' => ['manual' => (string) $manual],
+        'method' => __METHOD__
+      ]);
+
+      $startTimeStamp = $this->statusService->markInventoryStarted($fullInventory);
```

Comment:
Initially thought the variable name implied the start of the window but looks like it's the end. I would suggest renaming to avoid confusion.

Reply:
this is the start of the current sync - it is also the end of the window as we cannot look at changes that happened after we started syncing.

Reply:
I'm renaming the `endOfWindow` variable since it's used once in a tiny scope. The preexisting comment about the window should help.